src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -21,6 +21,16 @@ internal static class Utf8Encoder
         private const byte b1110_0000U = 0xE0; //224
         private const byte b1111_0000U = 0xF0; //240
         private const byte b1111_1000U = 0xF8; //248
+        
+        private const ushort Utf16HighSurrogateFirstCodePoint = 0xD800;
+        private const ushort Utf16HighSurrogateLastCodePoint = 0xDBFF;
```

Comment:
If these are the same constants that exist in the UTF16 encoder, moving them to a common place makes sense. I only put them in the UTF16 encoder because I was working at getting rid of the UnicodeCodePoint.

I suggest to put them as internal constants on UnicodeHelpers.cs and rename that class to EncodingHelpers or something like that. Then we can put all the constants used across classes there.